# 'likes' Report by Leah Fitzgerald 

In [1]:
import pandas as pd
import json
import matplotlib as plt

In [2]:
with open(r"C:\Users\leahf\OneDrive\Desktop\le.ah_20210902 (1)\likes\liked_posts.json") as p:
    dat = json.load(p)

Here I am importing the necessary packages for my code. In the 2nd cell, I am importing my personal Instagram file that I am focusing on for this assignment. 

In [3]:
dat.keys()

dict_keys(['likes_media_likes'])

In [4]:
dat['likes_media_likes'] 

[{'title': 'chezzwezz',
  'media_list_data': [],
  'string_list_data': [{'href': 'https://www.instagram.com/p/QmpWWGDWLZKuoy2M8D1mj5d7xiKPtZHMP_a_c0/',
    'value': 'ð\x9f\x91\x8d',
    'timestamp': 1349910284}]},
 {'title': 'queen_of_everaythang',
  'media_list_data': [],
  'string_list_data': [{'href': 'https://www.instagram.com/p/QS_WPmCswE/',
    'value': 'ð\x9f\x91\x8d',
    'timestamp': 1349222403}]},
 {'title': 'queen_of_everaythang',
  'media_list_data': [],
  'string_list_data': [{'href': 'https://www.instagram.com/p/QTFXfNis0V/',
    'value': 'ð\x9f\x91\x8d',
    'timestamp': 1349221988}]},
 {'title': 'queen_of_everaythang',
  'media_list_data': [],
  'string_list_data': [{'href': 'https://www.instagram.com/p/QJD5b6iszv/',
    'value': 'ð\x9f\x91\x8d',
    'timestamp': 1348884067}]},
 {'title': 'xxquotes14xx',
  'media_list_data': [],
  'string_list_data': [{'href': 'https://www.instagram.com/p/PnqhuTtBOs/',
    'value': 'ð\x9f\x91\x8d',
    'timestamp': 1347767079}]},
 {'tit

The third cell is helping me to determine the keys in my data. In the 4th, I am seeing the data that is inside of 'likes_media_likes'

In [5]:
df_likes = pd.DataFrame(dat['likes_media_likes'])

Here I am using the data from my file to create a data frame. I named it df_likes.

In [6]:
df_likes

title media_list_data  \
0                chezzwezz              []   
1     queen_of_everaythang              []   
2     queen_of_everaythang              []   
3     queen_of_everaythang              []   
4             xxquotes14xx              []   
...                    ...             ...   
5057          sydna_tygers              []   
5058            turnoverva              []   
5059        derrickarayven              []   
5060        sophia.stasiak              []   
5061           gretchab123              []   

                                       string_list_data  
0     [{'href': 'https://www.instagram.com/p/QmpWWGD...  
1     [{'href': 'https://www.instagram.com/p/QS_WPmC...  
2     [{'href': 'https://www.instagram.com/p/QTFXfNi...  
3     [{'href': 'https://www.instagram.com/p/QJD5b6i...  
4     [{'href': 'https://www.instagram.com/p/PnqhuTt...  
...                                                 ...  
5057  [{'href': 'https://www.instagram.com/p/B396oWq...  
5058  [{'href': 'https://www.instagram.com/p/B3mmbAz...  
5059  [{'href': 'https://www.instagram.com/p/B3mt-eK...  
5060  [{'href': 'https://www.instagram.com/p/B3k0gsd...  
5061  [{'href': 'https://www.instagram.com/p/B3gSIfg...  

[5062 rows x 3 columns]

In [8]:
days = [x['string_list_data'] for x in dat['likes_media_likes']]

In [9]:
days

[[{'href': 'https://www.instagram.com/p/QmpWWGDWLZKuoy2M8D1mj5d7xiKPtZHMP_a_c0/',
   'value': 'ð\x9f\x91\x8d',
   'timestamp': 1349910284}],
 [{'href': 'https://www.instagram.com/p/QS_WPmCswE/',
   'value': 'ð\x9f\x91\x8d',
   'timestamp': 1349222403}],
 [{'href': 'https://www.instagram.com/p/QTFXfNis0V/',
   'value': 'ð\x9f\x91\x8d',
   'timestamp': 1349221988}],
 [{'href': 'https://www.instagram.com/p/QJD5b6iszv/',
   'value': 'ð\x9f\x91\x8d',
   'timestamp': 1348884067}],
 [{'href': 'https://www.instagram.com/p/PnqhuTtBOs/',
   'value': 'ð\x9f\x91\x8d',
   'timestamp': 1347767079}],
 [{'href': 'https://www.instagram.com/p/PFW6GCKyMc/',
   'value': 'ð\x9f\x91\x8d',
   'timestamp': 1347671685}],
 [{'href': 'https://www.instagram.com/p/PBJ5rgrCHZ/',
   'value': 'ð\x9f\x91\x8d',
   'timestamp': 1346471223}],
 [{'href': 'https://www.instagram.com/p/PBJ69WLCHd/',
   'value': 'ð\x9f\x91\x8d',
   'timestamp': 1346471218}],
 [{'href': 'https://www.instagram.com/p/PBKAT2rCHm/',
   'value': 'ð

In [10]:
type(dat['likes_media_likes'])

list

In [11]:
dat['likes_media_likes'][7]

{'title': 'teenager_posts_tumblr',
 'media_list_data': [],
 'string_list_data': [{'href': 'https://www.instagram.com/p/PBJ69WLCHd/',
   'value': 'ð\x9f\x91\x8d',
   'timestamp': 1346471218}]}

Here I am pulling a specific value from my list. 

In [12]:
time = [p[0] for p in days]

Here I am creating 'time' to assist with creating 'timestamp' column.

In [13]:
time

[{'href': 'https://www.instagram.com/p/QmpWWGDWLZKuoy2M8D1mj5d7xiKPtZHMP_a_c0/',
  'value': 'ð\x9f\x91\x8d',
  'timestamp': 1349910284},
 {'href': 'https://www.instagram.com/p/QS_WPmCswE/',
  'value': 'ð\x9f\x91\x8d',
  'timestamp': 1349222403},
 {'href': 'https://www.instagram.com/p/QTFXfNis0V/',
  'value': 'ð\x9f\x91\x8d',
  'timestamp': 1349221988},
 {'href': 'https://www.instagram.com/p/QJD5b6iszv/',
  'value': 'ð\x9f\x91\x8d',
  'timestamp': 1348884067},
 {'href': 'https://www.instagram.com/p/PnqhuTtBOs/',
  'value': 'ð\x9f\x91\x8d',
  'timestamp': 1347767079},
 {'href': 'https://www.instagram.com/p/PFW6GCKyMc/',
  'value': 'ð\x9f\x91\x8d',
  'timestamp': 1347671685},
 {'href': 'https://www.instagram.com/p/PBJ5rgrCHZ/',
  'value': 'ð\x9f\x91\x8d',
  'timestamp': 1346471223},
 {'href': 'https://www.instagram.com/p/PBJ69WLCHd/',
  'value': 'ð\x9f\x91\x8d',
  'timestamp': 1346471218},
 {'href': 'https://www.instagram.com/p/PBKAT2rCHm/',
  'value': 'ð\x9f\x91\x8d',
  'timestamp': 1346

In [14]:
type(time[0])

dict

In [15]:
time = pd.DataFrame(time)

Here I am creating another column for 'timestamp'. 

In [16]:
time

href value   timestamp
0     https://www.instagram.com/p/QmpWWGDWLZKuoy2M8D...  👍  1349910284
1               https://www.instagram.com/p/QS_WPmCswE/  👍  1349222403
2               https://www.instagram.com/p/QTFXfNis0V/  👍  1349221988
3               https://www.instagram.com/p/QJD5b6iszv/  👍  1348884067
4               https://www.instagram.com/p/PnqhuTtBOs/  👍  1347767079
...                                                 ...   ...         ...
5057  https://www.instagram.com/p/B396oWqpTnOjt0kXeJ...  👍  1571885945
5058           https://www.instagram.com/p/B3mmbAzlyiM/  👍  1571079236
5059  https://www.instagram.com/p/B3mt-eKJ9Q6hOKp0UF...  👍  1571078177
5060  https://www.instagram.com/p/B3k0gsdl_lImxP9dUS...  👍  1571014507
5061  https://www.instagram.com/p/B3gSIfgphgxXG1QnXa...  👍  1570902977

[5062 rows x 3 columns]

In [17]:
df_likes.groupby('title').count()

media_list_data  string_list_data
title                                               
0fficial_addiea                  3                 3
10schmidt_g10                    1                 1
1_800_saintchiron                5                 5
1d_dailyupdates                 17                17
1domf                           22                22
...                            ...               ...
zaynstop                         3                 3
zeynepskaynak                    7                 7
zhxsgjn                          1                 1
zman2018                         1                 1
zzizziboy                        1                 1

[815 rows x 2 columns]

I used df_likes and grouped by 'title' in ascending order and included the count. 

In [18]:
df_likes = pd.DataFrame(y['title'] for y in dat['likes_media_likes'])

In [19]:
df_likes

0
0                chezzwezz
1     queen_of_everaythang
2     queen_of_everaythang
3     queen_of_everaythang
4             xxquotes14xx
...                    ...
5057          sydna_tygers
5058            turnoverva
5059        derrickarayven
5060        sophia.stasiak
5061           gretchab123

[5062 rows x 1 columns]

Here I pulled each occurance of 'title' with the row number. 

## Hypothesis & Questions

- When I first turned in this assignment, I forgot to explain what my code was doing. 
- Also, when reading through the Discord, I misunderstood and thought you wanted shortened versions.
- I also just googled 'nb viewer' and assumed there was only one. I ended up using the wrong one, I am sorry and it will not happen again. I bookmarked it :)

Who collected this data? 
   - Instagram (Facebook) collected it. 
       
Why was this data collected?
- To record the posts that I have interacted with in my time on instagram
       
In what ways may this data be reliable?
- It’s reliable because they record all of my interactions in chronological order.
- They need my interactions for advertising. 
- Also because it is the fact of every post that I have liked. 


In what ways may this data be unreliable?
- Development of my personal interests
- When I first started Instagram, I was 14 I think? I have changed very much and I am not interested in the same things I was   when I first made my account and first started interacting with posts. 

State a hypothesis and a data file that may help answer the hypothesis.
- Hypothesis: Instagram uses the information about the posts I have liked, to determine the content I see.  i.e. sponsored posts, the explore page, people to follow, and shopping interests.
- The statistical hypothesis would be that Instagram uses all 5061 of my likes to determine everything they need to know about me. 

A few data files that would help me confirm my hypothesis would be: 
- ads_interests, your_topics, your_reels_topics, or account_based_in